src/synapse/azext_synapse/custom.py

```diff
@@ -121,59 +123,126 @@ def create_workspace(cmd, client, resource_group_name, workspace_name, account_u
-        big_data_pool_info.auto_pause = AutoPauseProperties(enabled=auto_pause_enabled,
-                                                            delay_in_minutes=delay_in_minutes)
-
-    if library_requirements_filename or library_requirements_content:
-        big_data_pool_info.library_requirements = LibraryRequirements(filename=library_requirements_filename,
+
+    big_data_pool_info.auto_scale = AutoScaleProperties(enabled=enable_auto_scale, min_node_count=min_node_count,
+                                                        max_node_count=max_node_count)
+
+    big_data_pool_info.auto_pause = AutoPauseProperties(enabled=enable_auto_pause,
+                                                        delay_in_minutes=delay_in_minutes)
+
+    if library_requirements_file:
+        library_requirements_content = read_file_content(library_requirements_file)
+        big_data_pool_info.library_requirements = LibraryRequirements(filename=library_requirements_file,
                                                                       content=library_requirements_content)
     return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, workspace_name, spark_pool_name,
                        big_data_pool_info, force)
 
 
-def update_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name, tags=None):
-    return client.update(resource_group_name, workspace_name, spark_pool_name, tags)
+def update_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name,
+                      node_size=None, node_count=None, enable_auto_scale=None,
+                      min_node_count=None, max_node_count=None,
+                      enable_auto_pause=None, delay_in_minutes=None,
+                      library_requirements_file=None, tags=None, force=False, no_wait=False):
+    existing_spark_pool = client.get(resource_group_name, workspace_name, spark_pool_name)
+
+    if not existing_spark_pool:
+        raise CLIError("Failed to discover the spark pool {}".format(spark_pool_name))
+
+    if node_size:
+        existing_spark_pool.node_size = node_size
+    if node_count:
+        existing_spark_pool.node_count = node_count
+
+    if library_requirements_file:
+        library_requirements_content = read_file_content(library_requirements_file)
+        existing_spark_pool.library_requirements = LibraryRequirements(filename=library_requirements_file,
+                                                                       content=library_requirements_content)
+    if tags:
+        existing_spark_pool.tags = tags
+
+    if existing_spark_pool.auto_scale is not None:
+        if enable_auto_scale is not None:
+            existing_spark_pool.auto_scale.enabled = enable_auto_scale
+        if min_node_count:
+            existing_spark_pool.auto_scale.min_node_count = min_node_count
+        if max_node_count:
+            existing_spark_pool.auto_scale.min_node_count = max_node_count

```

Comment:
>min_node_count [](start = 43, length = 14)

max_node_count

Reply:
fixed

---
In reply to: [417128030](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r417128030) [](ancestors = 417128030)